mbed_tools_ci/assert_news.py

```diff
@@ -42,133 +40,64 @@ def validate_file_contents(self) -> None:
-        self.git.pull()
-        current_commit = self.git.get_current_commit()
-        # Delete `master` as it may already exist and be set to the one of interest (on CircleCI anyway and maybe on other systems)
-        if self.git.branch_exists(self.master_branch):
-            self.git.delete_branch(self.master_branch)
-        self.git.checkout(self.master_branch)
-        self.git.set_upstream_branch(self.master_branch)
-        self.git.pull()
-        master_branch_commit = self.git.get_current_commit()
-        self.git.checkout(self.current_branch)
-        project_root = configuration.get_value(
-            ConfigurationVariable.PROJECT_ROOT)
-        news_dir_relative_path = os.path.relpath(news_dir, start=project_root)
-        added_news = self.git.get_changes_list(
-            self.git.get_branch_point(
-                master_branch_commit, current_commit),
-            current_commit, change_type='a',
-            dir=news_dir_relative_path
-        )
-        extension_to_exclude = ['.toml', '.rst']
-        return [path for path in added_news if
-                len([ex for ex in extension_to_exclude if ex in path]) == 0]
-
-    def is_feature_branch(self) -> bool:
-        """States whether the current branch contains new features.
-
-        Returns:
-            True if current branch is a feature branch; False otherwise.
-        """
-        return not self.is_special_branch()
 
-    def is_special_branch(self) -> bool:
-        """Checks whether current branch is used for the release process.
-
-        Returns:
-              True if current branch is a special; False otherwise.
-        """
-        return (self.current_branch in [self.git.get_master_branch(),
-                                        self.git.get_beta_branch()]
-                ) or self.git.is_release_branch(self.current_branch)
-
-    def verify(self) -> None:
-        """Checks that news files were added in the current branch as part of the PR's changes.
-
-        The files are then individually checked in order to ensure
-        they follow the standard in terms of naming and content.
-
-        """
-        if not self.is_feature_branch():
-            logger.info(
-                f'No need for news file on branch [{self.current_branch}]'
-            )
-            return
-        added_news = self.find_news_file()
-        news_dir = configuration.get_value(ConfigurationVariable.NEWS_DIR)
-        project_root = configuration.get_value(
-            ConfigurationVariable.PROJECT_ROOT)
-        if not added_news or len(added_news) == 0:
-            raise FileNotFoundError(
-                f'PR must contain a news file in {news_dir}. See README.md'
-            )
-        logger.info(
-            f'{len(added_news)} new news files found in `{news_dir}`'
+def validate_news_files(git: GitWrapper, news_dir: str) -> None:
+    """Checks that news files exist and pass validation checks."""
```

Comment:
Could you document what the function parameters mean?
e.g. it is important to say than news_dir must be the relative path from project root

Reply:
Nice catch. Either that or I make it work on absolutes, all the way. What do you think? It'd require some additional logic resolving changes from git properly. 🤔